tutorials/tree/dictionary/main.cpp

```diff
@@ -0,0 +1,20 @@
+// ##########################################################################
+// # Alvaro Tolosa Delgado @ IFIC(Valencia,Spain)  alvaro.tolosa@ific.uv.es #
+// # Copyright (c) 2018 Alvaro Tolosa. All rights reserved.		 #
+// ##########################################################################
+
+#include "readTree.cpp"
+#include "writeTree.cpp"
```

Comment:
`#include`ing files that provide anything different from templated code, `inline` functions, or any other entities that are safe including from many translation units is, in general, frowned upon.

I would suggest instead to either:
- Directly include the code here (obviously removing the include guards and duplicated `#include` lines), or...
- Separately compile (as different translation units) each of those files and then link the resulting `.o` object files